website/docs/reference/dbt-jinja-functions/env_var.md

```diff
@@ -47,18 +47,32 @@ models:
 
  This can be useful to avoid compilation errors when the environment variable isn't available.
 
-### Special env var prefixes
+
+### Secrets
 
 <Changelog>
 
-  - **v0.19.0:** Introduced `DBT_ENV_CUSTOM_ENV_` prefix and artifact `metadata.env`
   - **v0.21.0:** Introduced `DBT_ENV_SECRET_` and log scrubbing
+  - **v1.0.0:** Restricted use of secret env vars to `profiles.yml` + `packages.yml`
+
+</Changelog>
+
+For certain configurations, you can use "secret" env vars. Any env var named with the prefix `DBT_ENV_SECRET_` will be:
+- Available for use in `profiles.yml` + `packages.yml`, via the same `env_var()` function
+- Disallowed everywhere else, including `dbt_project.yml` and model SQL
```

Comment:
Is this because the dbt_project.yml is available to the public? If so we might want to say it's to protect their secrets.

Reply:
I'll include a comment/justification